File: scripts/test-nominatim.mjs
// Test what Nominatim actually returns for Corrientes addresses
const BASE = "https://nominatim.openstreetmap.org/search?format=json&limit=10&countrycodes=ar&addressdetails=1"
const HEADERS = { "User-Agent": "GestionInmobiliariaRE/1.0" }

async function test(label, url) {
  console.log(`\n=== ${label} ===`)
  console.log("URL:", url)
  const res = await fetch(url, { headers: HEADERS })
  const data = await res.json()
  if (!data.length) {
    console.log("NO RESULTS")
    return
  }
  data.forEach((r, i) => {
    const a = r.address || {}
    console.log(`[${i}] lat=${r.lat} lon=${r.lon} type=${r.type} importance=${r.importance?.toFixed(3)}`)
    console.log(`     display_name=${r.display_name}`)
    console.log(`     addr.city="${a.city}" addr.town="${a.town}" addr.municipality="${a.municipality}" addr.village="${a.village}" addr.suburb="${a.suburb}" addr.county="${a.county}" addr.state="${a.state}"`)
  })
}

// Strategy 1: structured params
const params1 = new URLSearchParams({ street: "Uruguay 355", city: "Corrientes", state: "Corrientes", country: "Argentina" })
await test("Structured: street+city+state", `${BASE}&${params1}`)

await new Promise(r => setTimeout(r, 1000))

// Strategy 2: free text full
const q2 = encodeURIComponent("Uruguay 355, Corrientes, Corrientes, Argentina")
await test("Free text: street+city+state+country", `${BASE}&q=${q2}`)

await new Promise(r => setTimeout(r, 1000))

// Strategy 3: free text city only
const q3 = encodeURIComponent("Uruguay 355, Corrientes, Argentina")
await test("Free text: street+city+country", `${BASE}&q=${q3}`)

File: scripts/test-nominatim.py
import urllib.request
import json

headers = {"User-Agent": "GestionInmobiliariaRE/1.0"}

def fetch(url):
    req = urllib.request.Request(url, headers=headers)
    with urllib.request.urlopen(req) as resp:
        return json.loads(resp.read())

# Test 1: Nominatim searching just the city centroid
print("=== Nominatim: ciudad capital Corrientes (centroide) ===")
url = "https://nominatim.openstreetmap.org/search?format=json&limit=5&addressdetails=1&q=Corrientes,+Corrientes,+Argentina&countrycodes=ar"
data = fetch(url)
for i, r in enumerate(data[:5]):
    addr = r.get("address", {})
    print(f"  [{i}] lat={r['lat']} lon={r['lon']} type={r.get('type')}")
    print(f"       display={r['display_name'][:100]}")

# Test 2: GeoRef localidades - find the centroid of city Corrientes Capital
print("\n=== GeoRef: localidades de Corrientes ===")
url2 = "https://apis.datos.gob.ar/georef/api/localidades?nombre=Corrientes&provincia=Corrientes&max=5&campos=completo"
data2 = fetch(url2)
locs = data2.get("localidades", [])
for i, r in enumerate(locs):
    ub = r.get("centroide", {})
    print(f"  [{i}] lat={ub.get('lat')} lon={ub.get('lon')}")
    print(f"       nombre={r.get('nombre')} | prov={r.get('provincia',{}).get('nombre')}")
